oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -765,9 +810,33 @@ int mca_spml_ucx_fence(shmem_ctx_t ctx)
 
 int mca_spml_ucx_quiet(shmem_ctx_t ctx)
 {
+    int flush_get_data;
     int ret;
+    size_t i, j, nprocs;
     mca_spml_ucx_ctx_t *ucx_ctx = (mca_spml_ucx_ctx_t *)ctx;
 
+    if (mca_spml_ucx.synchronized_quiet) {
+        nprocs = oshmem_num_procs();
+        for (i = 0; i < nprocs; i += SSHMEM_MASK_BITS) {
+            if (!ucx_ctx->put_op_mask[SSHMEM_MASK_IDX(i)]) {
+                continue;
+            }
+
+            for (j = 0; (j < SSHMEM_MASK_BITS) && (i + j < nprocs); j++) {
+                if (SSHMEM_IS_BIT(ucx_ctx->put_op_mask, i + j)) {
```

Comment:
go over only relevant procs: if the mask bit was not set, add proc number to an array

Reply:
ok, done